mantidproject/mantid
# Discrete integration slider in the instrument viewer

**Description of work.**
This PR modifies the integration slider of the instrument viewer so that if the bin axis of the workspace is discrete, the slider is not continuous but works in discrete intervals based on the indices of the workspace's bins. 


**To test:**
-  Load `./ExternalData/Testing/Data/UnitTest/ILL/D11B/kinetic.nxs` and open the instrument viewer.
- Check the slider goes from 0 to 2.
- Try moving the slider, reducing and increasing its size, both with the slider and the spin boxes, and both with mouse and keyboard,  and check the view is updated accordingly. 


WIP for #31943. 






---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
